### PR TITLE
Update IDEA project files

### PR DESCRIPTION
**Pull request by stevemessick**
@devoncarew @pq 

Updates project files for IntelliJ and Android Studio that are generated by 'flutter create'. For all three project types:
 * Open the Project view (not Android, which is default for AS)
 * Mark special directories
 * Do not generate any balloons when opening the project
 * Add the Android facet if needed
 * Define the project SDK if needed

### CODE REVIEW

**Comment by stevemessick** [comment]
Forgot to mention that this fixes a couple plugin issues:
https://github.com/flutter/flutter-intellij/issues/1347
https://github.com/flutter/flutter-intellij/issues/1343

File: packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <project version="4">
   <component name="FileEditorManager">
     <leaf>
-      <file leaf-file-name="main.dart" pinned="false" current-in-tab="true">
+      <file leaf-file-name="{{projectName}}.dart" pinned="false" current-in-tab="true">
```
Comment (mit-mit) [inline packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: Isn't the file still called main.dart in any project from the app template?

File: packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl [hunk 2]
```diff
@@ -11,7 +11,7 @@
           </provider>
         </entry>
       </file>
-      <file leaf-file-name="main.dart" pinned="false" current-in-tab="true">
+      <file leaf-file-name="main.dart" pinned="false" current-in-tab="false">
```
Comment (mit-mit) [inline packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: Does this need to be projevtname.dart in packages?

File: packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl (hunk 1 above)
Comment (stevemessick) [reply packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: Yes, but this is for plugins. See below for discussion.

File: packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl (hunk 2 above)
Comment (stevemessick) [reply packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: In packages and apps we only open one file. This is for plugins, where we open two files. However, two files get opened only when a project is created by the command-line tool then opened in IntelliJ. If you create the project using the IntelliJ wizard only one file gets opened. That's not a regression but is a bug we should try to fix.

Comment (stevemessick) [reply packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: Filed https://github.com/flutter/flutter-intellij/issues/1370

Comment (mit-mit) [reply packages/flutter_tools/templates/plugin/.idea/workspace.xml.tmpl]: Got it, thanks

**Review by pq — APPROVED** [review]
LGTM

**Pull request merged**